segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/DebugBookKeeperLogWrapper.java

```diff
@@ -151,16 +151,19 @@ public void disable() throws DurableDataLogException {
      * @param candidateLedgers A List of {@link LedgerHandle}s that contain all the Ledgers that this {@link BookKeeperLog}
      *                         should contain. This could be the list of all BookKeeper Ledgers or a subset, as long as
      *                         it contains all Ledgers that list this {@link BookKeeperLog} as their owner.
+     * @param overwrite  Boolean denoting whether reconcile has to be forced even if log is enabled
      * @return True if something changed (and the metadata is updated), false otherwise.
      * @throws IllegalStateException   If this BookKeeperLog is not disabled.
      * @throws DurableDataLogException If an exception occurred while updating the metadata.
      */
-    public boolean reconcileLedgers(List<? extends ReadHandle> candidateLedgers) throws DurableDataLogException {
+    public boolean reconcileLedgers(List<? extends ReadHandle> candidateLedgers, boolean overwrite) throws DurableDataLogException {
         // Load metadata and verify if disabled (metadata may be null if it doesn't exist).
         LogMetadata metadata = this.log.loadMetadata();
         final long highestLedgerId;
-        if (metadata != null) {
+        if (!overwrite) {
```

Comment:
This is changing the behavior of this method in 3 different ways:
1. Now the check if metadata is only done when we execute this method without overwrite.
2. The precondition check may throw a `NullPointerException`, as it is invoking `metadata.isEnabled()` and `metadata` can be `null`.
3. If `metadata != null`, there is no such check.

Reply:
reverted the change related to this method.